GETTING_STARTED_CLI.markdown

```diff
@@ -0,0 +1,318 @@
+  computation.
+- `vc-fee`/`freestanding-execution-environment` - Provides a freestanding
+  execution environment that can be used to test Veracruz programs without
+  needing the full attestation/TEE framework.
+- `vc-wc`/`wasm-checker` - Checks that a Veracruz program is able to run in
+  a Veracruz computation.
+
+## Building the example
+
+For this walkthrough we're going to use the Shamir secret sharing example,
+which can be found in the `sdk/rust-examples/shamir-secret-sharing`
+directory.
+
+We need to use Xargo to cross-compile this example into WebAssembly, fortunately
+this is all captured in the example's Makefile:
+
+``` bash
+$ make -C sdk/rust-examples/shamir-secret-sharing
+```
+
+You should now find the example compiled to WebAssembly in the `target/wasm32-wasi/release`
+directory. This will be our program to execute inside a Veracruz enclave:
+
+``` bash
+$ ls sdk/rust-examples/shamir-secret-sharing/target/wasm32-wasi/release/shamir-secret-sharing.wasm
+sdk/rust-examples/shamir-secret-sharing/target/wasm32-wasi/release/shamir-secret-sharing.wasm
+```
+
+Lets go ahead and copy this to an example directory to make the paths a bit
+easier to use:
+
+``` bash
+$ mkdir -p example
+$ cp sdk/rust-examples/shamir-secret-sharing/target/wasm32-wasi/release/shamir-secret-sharing.wasm example/example-binary.wasm
+```
+
+<!-- TODO run example with freestanding-execution-environment? -->
+
+## Generating certificates
+
+Identities in Veracruz are specified by a private RSA key and signed
+x509 certificate.
+
+In practice, a single identity can have many roles, but for our example
+we're going to create a separate identity for the program provider, result
+reader, and three data providers:
+
+``` bash
+$ openssl genrsa -out example/example-program-key.pem 2048
+$ openssl req -new -x509 -sha256 -nodes -days 3650 \
+    -key example/example-program-key.pem \
+    -out example/example-program-cert.pem \
+    -config test-collateral/cert.conf
+
+$ openssl genrsa -out example/example-data0-key.pem 2048
+$ openssl req -new -x509 -sha256 -nodes -days 3650 \
+    -key example/example-data0-key.pem \
+    -out example/example-data0-cert.pem \
+    -config test-collateral/cert.conf
+
+$ openssl genrsa -out example/example-data1-key.pem 2048
+$ openssl req -new -x509 -sha256 -nodes -days 3650 \
+    -key example/example-data1-key.pem \
+    -out example/example-data1-cert.pem \
+    -config test-collateral/cert.conf
+
+$ openssl genrsa -out example/example-data2-key.pem 2048
+$ openssl req -new -x509 -sha256 -nodes -days 3650 \
+    -key example/example-data2-key.pem \
+    -out example/example-data2-cert.pem \
+    -config test-collateral/cert.conf
+
+$ openssl genrsa -out example/example-result-key.pem 2048
+$ openssl req -new -x509 -sha256 -nodes -days 3650 \
+    -key example/example-result-key.pem \
+    -out example/example-result-cert.pem \
+    -config test-collateral/cert.conf
+```
+
+<!-- TODO generate CA cert?
```

Comment:
What is the status of this TODO?

Reply:
Ah I forgot to bring this up last meeting, I don't know how to generate a cert that can be used for the CA authority, so I'm currently using the hard-coded CA cert used by the tests. It's not super important, I was just wanting GETTING_STARTED_CLI.md to contain all the commands necessary to set up your own computation (populate_test_database.sh, pgen, etc).

Now that @dreemkiller is back though, this should be easy to add.

Reply:
To generate the key:
```
openssl ecparam -name prime256v1 -genkey -noout -out CAkey.pem
```
To generate the CA certificate:
```
openssl req -x509 -new -nodes -key CAkey.pem -sha256 -days 1825 -subj "/CN=VeracruzProxyServer/OU=Proxy/O=Veracruz/L=Austin/ST=Texas/C=US" -out CACert.pem
```

Reply:
This should now be working 👍